ui/app/pages/create-account/connect-hardware/index.js

```diff
@@ -139,11 +139,24 @@ class ConnectHardwareForm extends Component {
           this.setState({ browserSupported: false, error: null });
         } else if (errorMessage.includes(U2F_ERROR)) {
           this.setState({ error: U2F_ERROR });
+        } else if (
+          errorMessage === 'LEDGER_LOCKED' ||
+          errorMessage === 'LEDGER_WRONG_APP'
+        ) {
+          this.setState({
+            error: this.context.t('ledgerLocked'),
```

Comment:
Huh. So I tried using the Ledger Live integration, and predictably it didn't work (the `ledger-live://` protocol isn't recognized on Linux), but I didn't see a timeout error either. It kept trying to connect forever.

Reply:
With U2F, the same timeout should be used -- nothing changes here.  The timeout for ledger live is 3 minutes -- support asked for a larger amount of time: (https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/54/files#diff-c0bdedb407c27ec7b9cc5c2e4a8d1d9a041eafa4a92c66b43f70c6e1c234af45R13)

With regard too `Internal JSON-RPC error`, that's not desirable; error messages shouldn't have changed.  I tested on our current released production version of MM (9.4.0) and I see the appropriate error when the Ledger is connected to my computer but not unlocked.  I checked out current `develop`, without my `eth-ledger-bridge-keyring` version update and the rest of the Ledger Live stuff, and also see "Internal JSON-RPC error." when the Ledger isn't unlocked.  

Could you confirm the same?  Any ideas on how we'd get here?

Reply:
My update to `gh-pages` is hitting the correct error:

<img width="1382" alt="WrongAppErrorReturned" src="https://user-images.githubusercontent.com/46655/115760039-88ad9100-a366-11eb-922e-a56d4e6567f0.png">

Reply:
Interesting. I'll give this another try!

Reply:
I tested on `develop` and saw the same "forever loading" when trying to connect via U2F. So that appears to be pre-existing at least. Similarly, when trying to switch pages with the Ethereum app closed, I see the "Internal JSON-RPC error" as well.